library/src/main/java/org/apache/apex/malhar/lib/wal/FileSystemWAL.java

```diff
@@ -680,9 +681,7 @@ protected void flush() throws IOException
           //until the stream is closed on the local FS, readers don't see any data.
           close();
         } else {
-          Syncable syncableOutputStream = (Syncable)outputStream;
-          syncableOutputStream.hflush();
-          syncableOutputStream.hsync();
+          ((HdfsDataOutputStream)outputStream).hsync(EnumSet.of(SyncFlag.UPDATE_LENGTH));
```

Comment:
Looks like you need this to address an issue you've uncovered that relates to the original problem explained in the JIRA of not getting the correct largest recovery window. Could you explain what that issue is exactly? Also if including the functionality provide by the dependency is the way to do this, please add a comment in the code describing the problem, in case in the future there is a substitute in the common api.